src/main/java/com/github/ermay12/regex/Regex.java

```diff
@@ -241,6 +244,7 @@
   private Pattern pattern;
   private String rawRegex;
   private final String privatesyncobj = "";
+  protected Map<CapturingGroup, Integer> groupToIndex = new HashMap<>();
```

Comment:
Some undocumented behavior with this - what happens when you use the same capturing group multiple times in the same regex?

Reply:
Changes are up for this